File: backend/app/schemas/phase.py
from pydantic import BaseModel


class PhaseMetric(BaseModel):
    label: str
    count: int


class PhaseConfiguration(BaseModel):
    ebay_api_credentials_configured: bool
    ebay_compliance_configured: bool


class PhaseStatusRead(BaseModel):
    phase: str
    status: str
    core_ready: bool
    database: str
    metrics: list[PhaseMetric]
    ready_checks: dict[str, bool]
    configuration: PhaseConfiguration
    pending_configuration: list[str]

File: frontend/src/types/phase.ts
export interface PhaseMetric {
  label: string
  count: number
}

export interface PhaseConfiguration {
  ebay_api_credentials_configured: boolean
  ebay_compliance_configured: boolean
}

export interface PhaseStatus {
  phase: string
  status: string
  core_ready: boolean
  database: string
  metrics: PhaseMetric[]
  ready_checks: Record<string, boolean>
  configuration: PhaseConfiguration
  pending_configuration: string[]
}
